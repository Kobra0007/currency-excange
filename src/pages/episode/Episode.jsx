import { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import { episodeFetchById } from '../../redux/actions/episode'
import { getLoading, getEpisode } from '../../redux/selectors/episode'

import './episode.scss'

export default function Episode(props) {
  const {
    match: {
      params: { episodeId },
    },
  } = props

  const isLoading = useSelector(getLoading)
  const episode = useSelector(getEpisode)

  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(episodeFetchById({ episodeId }))
  }, [dispatch, episodeId])

  return (
    <div className="l-container-episode">
      {!isLoading ? (
        <>
          <header className="l-container-episode__header">
            <h1>{episode.name}</h1>
          </header>
          <main className="l-container-episode__main">
            <section className="c-info-episode">
              <figure className="c-image-episode">
                <img src={episode.image?.medium} alt={episode.name} />
              </figure>
              <div
                className="c-summary-episode"
                // eslint-disable-next-line react/no-danger
                dangerouslySetInnerHTML={{ __html: episode.summary }}
              />
            </section>
          </main>
        </>
      ) : (
        <p>Loading...</p>
      )}
    </div>
  )
}

Episode.defaultProps = {
  match: {},
}

Episode.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      episodeId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    }),
  }),
}
