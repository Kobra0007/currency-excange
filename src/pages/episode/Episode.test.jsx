import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import Episode from './Episode'

const props = {
  match: {
    params: {
      episodeId: 4,
    },
  },
}
describe('Episode.jsx', () => {
  describe('<Episode />', () => {
    const initialState = {
      episode: {
        error: null,
        isLoading: false,
        episodes: [],
        episodeInfo: {},
      },
      show: {
        error: null,
        isLoading: false,
        showInfo: {},
      },
    }

    const mockStore = configureStore()
    test('should render and match shapshot', () => {
      const component = shallow(
        <Provider store={mockStore(initialState)}>
          <Episode {...props} />
        </Provider>,
      )

      expect(component).toMatchSnapshot()
    })
  })
})
