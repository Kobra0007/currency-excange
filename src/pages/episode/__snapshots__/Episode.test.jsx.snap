// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode.jsx <Episode /> should render and match shapshot 1`] = `ShallowWrapper {}`;
